mobileApp/lib/ui/IntroFlow/authFlow/namePage.dart

```diff
@@ -3,39 +3,43 @@ import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
 
 // 📦 Package imports:
-import 'package:provider/provider.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
 
 // 🌎 Project imports:
 import 'package:YouOweMe/ui/IntroFlow/loginUser.dart';
+import 'package:YouOweMe/ui/IntroFlow/providers.dart';
 
-class NamePage extends StatefulWidget {
+class NamePage extends StatefulHookWidget {
   @override
   _NamePageState createState() => _NamePageState();
 }
 
 class _NamePageState extends State<NamePage> {
   final TextEditingController nameController = TextEditingController();
+
   @override
   void initState() {
     super.initState();
     nameController.addListener(() => setName());
   }
 
   void setName() {
+    LoginUser introFlowUser = introFlowUserProvider.read(context);
     String userName = nameController.text;
-    Provider.of<LoginUser>(context, listen: false).addName(userName);
+    introFlowUser.addName(userName);
```

Comment:
Test this once again, to see it does not break when used outside the build method.